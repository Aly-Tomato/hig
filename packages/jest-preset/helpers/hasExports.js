module.exports = function hasExports(index, cases) {
  cases.forEach(({ name, value }) => {
    it(`exports ${name}`, () => {
      expect(index).toHaveProperty(name, value);
    });
  });
};
