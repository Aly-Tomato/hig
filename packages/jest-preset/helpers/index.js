/* eslint-disable global-require */
module.exports = {
  hasExports: require("./hasExports"),
  lastCallOfMock: require("./lastCallOfMock"),
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
