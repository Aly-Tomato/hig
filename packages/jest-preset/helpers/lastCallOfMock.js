module.exports = function lastCallOfMock(mockFn) {
  return mockFn.mock.calls[mockFn.mock.calls.length - 1];
};
