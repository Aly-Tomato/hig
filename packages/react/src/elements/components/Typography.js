import React from "react";
import PropTypes from "prop-types";
import Typography from "../../adapters/TypographyAdapter";

export function H1({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="h1"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function H2({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="h2"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function H3({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="h3"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Text({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="text"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Sub1({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="sub1"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Sub2({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="sub2"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Body({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="body"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Bold({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="bold"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Disabled({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="disabled"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

export function Caption({ children, size, bold, color, disabled }) {
  return (
    <Typography
      type="caption"
      bold={bold}
      size={size}
      color={color}
      disabled={disabled}
      text={children}
    />
  );
}

const docgenInfo = {
  props: {
    bold: {
      description: "Whether to render bold text"
    },
    color: {
      description:
        "A text color. One of: 'hig-white', 'hig-cool-gray-70', 'hig-blue-50', 'hig-green-good', 'hig-yellow-warning', 'hig-red-alert'"
    },
    disabled: {
      description: "Whether to show text as disabled"
    },
    size: {
      description: "One of: 'small', 'medium', 'large'"
    },
    children: {
      description: "any content"
    }
  }
};

const propTypes = {
  /**
   * Whether to render bold text
   */
  bold: PropTypes.bool,
  /**
   * A color to apply to the containing text
   */
  color: PropTypes.oneOf([
    "hig-white",
    "hig-cool-gray-70",
    "hig-blue-50",
    "hig-green-good",
    "hig-yellow-warning",
    "hig-red-alert"
  ]),
  /**
   * Whether to show text as disabled
   */
  disabled: PropTypes.bool,
  /**
   * A size to apply to the containing text
   */
  size: PropTypes.oneOf(["small", "medium", "large"]),
  /**
   * Text to render
   */
  children: PropTypes.node
};

[H1, H2, H3, Text, Sub1, Sub2, Body, Bold, Disabled, Caption].forEach(
  componentClass => {
    componentClass.__docgenInfo = docgenInfo; // eslint-disable-line no-param-reassign
    componentClass.propTypes = propTypes; // eslint-disable-line no-param-reassign
  }
);
