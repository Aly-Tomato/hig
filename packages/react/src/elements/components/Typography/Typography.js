import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import "./typography.scss";

export const validColors = [
  "hig-white",
  "hig-cool-gray-70",
  "hig-blue-50",
  "hig-green-good",
  "hig-yellow-warning",
  "hig-red-alert"
];

export const validSizes = ["small", "medium", "large"];

export const validTypes = [
  "h1",
  "h2",
  "h3",
  "text",
  // Types below intended to be deprecated
  "sub1",
  "sub2",
  "body",
  "bold",
  "disabled",
  "caption"
];

export default class Typography extends Component {
  render() {
    const classes = cx("hig__typography", {
      [`hig__typography__${this.props.type}`]: this.props.type,
      [`hig__typography--${this.props.color}`]: this.props.color,
      [`hig__typography--${this.props.size}`]: this.props.size,
      "hig__typography--bold": this.props.bold,
      "hig__typography--disabled": this.props.disabled
    });

    /**
     * N.B.: The intent is to apply opacity to the color. Typography components currently only render text, so
     * applying opacity should have no side effects. If this component ever wraps styled content, we should reconsider
     * this mechanism, because opacity in children nodes can be magnified by this parent node.
     */
    const customStyles =
      typeof this.props.opacity === "number"
        ? { opacity: this.props.opacity }
        : {};

    return (
      <span className={classes} style={customStyles}>
        {this.props.text}
      </span>
    );
  }
}

Typography.propTypes = {
  /**
   * Whether to render bold text
   */
  bold: PropTypes.bool,
  /**
   * Colors the text with one of the supported HIG colors
   */
  color: PropTypes.oneOf(validColors),
  /**
   * Whether to show text as disabled
   */
  disabled: PropTypes.bool,
  /**
   * An opacity value to modify the color, between 0.0 and 1.0
   */
  opacity: PropTypes.number,
  /**
   * Sizes the text with one of the supported modifiers
   */
  size: PropTypes.oneOf(validSizes),
  /**
   * Indicates the initial Typography style
   */
  type: PropTypes.oneOf(validTypes).isRequired,
  /**
   * Text to render styled based on provided type
   */
  text: PropTypes.string.isRequired
};
