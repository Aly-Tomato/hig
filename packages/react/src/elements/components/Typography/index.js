export { default as H1 } from "./H1";
export { default as H2 } from "./H2";
export { default as H3 } from "./H3";
export { default as Text } from "./Text";

export { default as Body } from "./Body";
export { default as Bold } from "./Bold";
export { default as Caption } from "./Caption";
export { default as Disabled } from "./Disabled";
export { default as Sub1 } from "./Sub1";
export { default as Sub2 } from "./Sub2";

export const _VALID_COLORS = Object.freeze([
  "hig-white",
  "hig-cool-gray-70",
  "hig-blue-50",
  "hig-green-good",
  "hig-yellow-warning",
  "hig-red-alert"
]);

export const _VALID_SIZES = Object.freeze(["small", "medium", "large"]);

export const _VALID_TYPES = Object.freeze([
  "h1",
  "h2",
  "h3",
  "text",
  // Types below intended to be deprecated
  "sub1",
  "sub2",
  "body",
  "bold",
  "disabled",
  "caption"
]);
