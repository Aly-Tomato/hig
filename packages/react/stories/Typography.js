import React from "react";
import { storiesOf } from "@storybook/react";
import { Typography as VanillaTypography } from "hig-vanilla";
import { number, boolean, select } from "@storybook/addon-knobs/react";
import {
  H1,
  H2,
  H3,
  Text,
  Body,
  Bold,
  Caption,
  Disabled,
  Sub1,
  Sub2
} from "../src/hig-react";

[H1, H2, H3, Text, Body, Bold, Caption, Disabled, Sub1, Sub2].forEach(
  Component => {
    const name = Component.name;

    storiesOf(name, module).add("with text", () => (
      <Component
        bold={boolean("Bold", false)}
        disabled={boolean("Disabled", false)}
        color={select(
          "Color",
          VanillaTypography.VALID_COLORS,
          "hig-cool-gray-70"
        )}
        opacity={number("Opacity", 1.0, {
          range: true,
          min: 0.0,
          max: 1.0,
          step: 0.1
        })}
        size={select("Size", VanillaTypography.VALID_SIZES, "medium")}
      >
        {`${name} example text`}
      </Component>
    ));
  }
);
