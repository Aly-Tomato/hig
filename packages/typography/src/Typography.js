import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { css } from "emotion";
import { ThemeContext } from "@hig/theme-context";

import {
  _VALID_COLORS,
  _VALID_SIZES,
  _VALID_TYPES,
  AVAILABLE_ALIGNMENTS,
  AVAILABLE_FONT_WEIGHTS,
  AVAILABLE_VARIANTS
} from "./_constants";
import "./typography.scss";
import stylesheet from "./Typography.stylesheet";

export default class Typography extends Component {
  static propTypes = {
    /**
     * Sets the horizontal alignment of an inline box.
     */
    align: PropTypes.oneOf(AVAILABLE_ALIGNMENTS),
    /**
     * Whether to render bold text
     */
    bold: PropTypes.bool,
    /**
     * Text or content to render.
     */
    children: PropTypes.node,
    /**
     * Colors the text with one of the supported HIG colors
     */
    color: PropTypes.oneOf(_VALID_COLORS),
    /**
     * Whether to show text as disabled
     */
    disabled: PropTypes.bool,
    /**
     * Specifies the weight (or boldness) of the font.
     */
    fontWeight: PropTypes.oneOf(AVAILABLE_FONT_WEIGHTS),
    /**
     * An opacity value to modify the color, between 0.0 and 1.0
     */
    opacity: PropTypes.number,
    /**
     * Sizes the text with one of the supported modifiers
     */
    size: PropTypes.oneOf(_VALID_SIZES),
    /**
     * Indicates the initial Typography style
     */
    type: PropTypes.oneOf(_VALID_TYPES),
    /**
     * Text to render styled based on provided type
     */
    text: PropTypes.string,
    /**
     * Indicates the initial Typography style
     */
    variant: PropTypes.oneOf(AVAILABLE_VARIANTS)
  };

  classes = resolvedRoles =>
    cx(
      "hig__typography",
      {
        [`hig__typography__${this.props.type}`]: this.props.type,
        [`hig__typography--${this.props.color}`]: this.props.color,
        [`hig__typography--${this.props.size}`]: this.props.size,
        "hig__typography--bold": this.props.bold,
        "hig__typography--disabled": this.props.disabled
      },
      css(stylesheet(this.props, resolvedRoles).typography)
    );

  /**
   * N.B.: The intent is to apply opacity to the color. Typography components currently only render text, so
   * applying opacity should have no side effects. If this component ever wraps styled content, we should reconsider
   * this mechanism, because opacity in children nodes can be magnified by this parent node.
   */
  opacity = () =>
    typeof this.props.opacity === "number"
      ? { opacity: this.props.opacity }
      : {};

  render() {
    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => (
          <span className={this.classes(resolvedRoles)} style={this.opacity()}>
            {this.props.text}
            {this.props.children}
          </span>
        )}
      </ThemeContext.Consumer>
    );
  }
}
