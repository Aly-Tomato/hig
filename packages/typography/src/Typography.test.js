import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import Typography from "./Typography";

describe("Typography", () => {
  function newProps(variant, ...args) {
    const variantDetails = variant ? `${variant} variant` : "Default";
    const otherDetails = args.length ? ` with ${args.join(" ")}` : "";
    return {
      variant,
      children: `${variantDetails} should render nicely${otherDetails}.`
    };
  }

  function oldProps(type, ...args) {
    const otherDetails = args.length ? ` with ${args.join(" ")}` : "";
    return {
      type,
      text: `${type} type should render nicely${otherDetails}.`
    };
  }

  takeSnapshotsOf(Typography, [
    // new API tests
    { description: "renders default Typography", props: newProps() },
    {
      description: "renders Typography with align and fontWeight props",
      props: Object.assign(
        newProps("body", "align center and fontWeight bold"),
        { align: "center", fontWeight: "bold" }
      )
    },
    {
      description: "renders body variant Typography",
      props: newProps("body")
    },
    {
      description: "renders caption variant Typography",
      props: newProps("caption")
    },
    {
      description: "renders h1 variant Typography",
      props: newProps("h1")
    },
    {
      description: "renders h2 variant Typography",
      props: newProps("h2")
    },
    {
      description: "renders h3 variant Typography",
      props: newProps("h3")
    },

    // deprecated API tests
    {
      description: "renders text",
      props: oldProps("text")
    },
    {
      description: "renders with a type",
      props: oldProps("h1")
    },
    {
      description: "renders bold",
      props: Object.assign(oldProps("text", "bold"), { bold: true })
    },
    {
      description: "renders disabled",
      props: Object.assign(oldProps("text", "disabled"), { disabled: true })
    },
    {
      description: "renders with a size",
      props: Object.assign(oldProps("text", "size small"), { size: "small" })
    },
    {
      description: "renders with a color",
      props: Object.assign(oldProps("text", "color hig-cool-gray-70"), {
        color: "hig-cool-gray-70"
      })
    },
    {
      description: "renders with an opacity",
      props: Object.assign(oldProps("text", "opacity 0.3"), { opacity: 0.3 })
    }
  ]);
});
