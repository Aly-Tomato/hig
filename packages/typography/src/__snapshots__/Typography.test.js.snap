// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography  1`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  Default should render nicely.
</span>
`;

exports[`Typography  2`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  body variant should render nicely with align center and fontWeight bold.
</span>
`;

exports[`Typography  3`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  body variant should render nicely.
</span>
`;

exports[`Typography  4`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  caption variant should render nicely.
</span>
`;

exports[`Typography  5`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  h1 variant should render nicely.
</span>
`;

exports[`Typography  6`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  h2 variant should render nicely.
</span>
`;

exports[`Typography  7`] = `
<span
  className="hig__typography css-0"
  style={Object {}}
>
  h3 variant should render nicely.
</span>
`;

exports[`Typography  8`] = `
<span
  className="hig__typography hig__typography__text css-0"
  style={Object {}}
>
  text type should render nicely.
</span>
`;

exports[`Typography  9`] = `
<span
  className="hig__typography hig__typography__h1 css-0"
  style={Object {}}
>
  h1 type should render nicely.
</span>
`;

exports[`Typography  10`] = `
<span
  className="hig__typography hig__typography__text hig__typography--bold css-0"
  style={Object {}}
>
  text type should render nicely with bold.
</span>
`;

exports[`Typography  11`] = `
<span
  className="hig__typography hig__typography__text hig__typography--disabled css-0"
  style={Object {}}
>
  text type should render nicely with disabled.
</span>
`;

exports[`Typography  12`] = `
<span
  className="hig__typography hig__typography__text hig__typography--small css-0"
  style={Object {}}
>
  text type should render nicely with size small.
</span>
`;

exports[`Typography  13`] = `
<span
  className="hig__typography hig__typography__text hig__typography--hig-cool-gray-70 css-0"
  style={Object {}}
>
  text type should render nicely with color hig-cool-gray-70.
</span>
`;

exports[`Typography  14`] = `
<span
  className="hig__typography hig__typography__text css-0"
  style={
    Object {
      "opacity": 0.3,
    }
  }
>
  text type should render nicely with opacity 0.3.
</span>
`;
